data/tutorials/language/1ds_01_arrays.md

```diff
@@ -0,0 +1,158 @@
+
+```ocaml
+# Array.map (fun x -> x * x) even_numbers;;
+- : int array = [|0; 4; 1764; 36; 64|]
+```
+
+### Folding an Array
+
+To combine all the elements of an array into a single result, we can use the `Array.fold_left` and `Array.fold_right` functions. These functions take a binary function, an initial accumulator value, and an array as arguments. The binary function takes two arguments: the current value of the accumulator and the current element of the array and returns a new accumulator value. Both functions traverse the array
+ but in opposite directions. This is essentially the same as `List.fold_left` and `List.fold_right`.
+
+Here is the signature of `Array.fold_left`:
+
+```ocaml
+val fold_left : ('a -> 'b -> 'a) -> 'a -> 'b array -> 'a
+```
+
+`fold_left f init a` computes `f (... (f(f init a.(0)) a.(1)) ...) a.(n-1)`
+
+Similarly, we can use the `Array.fold_right` function, which switches the order of its arguments:
+
+```ocaml
+val fold_right : ('b -> 'a -> 'a) -> 'b array -> 'a -> 'a
+```
+
+`fold_right f a init` computes `f a.(0) (f a.(1) ( ... (f a.(n-1) init) ...))`
+
+These functions derive a single value from the whole array. For example, they can be used to find the maximum element of an array:
+
+```ocaml
+# Array.fold_left max min_int even_numbers;;
+- : int = 42
+```
+
+Where `max` is the maximum function defined on elements of type int.
+
+### Sorting an Array
+
+To sort an array, we can use the `Array.sort` function. This function takes as arguments:
+- a comparison function
+- an array
+Ir sorts the provided array in place, in ascending order, according to the provided comparison function. Sorting performed by `Array.sort` modifies the content of the provided array, that why it returns `unit`. For example, to sort the array `even_numbers` created above, we can use:
+
+```ocaml
+# Array.sort compare even_numbers;;
+- : unit = ()
+# even_numbers;;
+- : int array = [|0; 2; 6; 8; 42|]
+```
+
+## Copying Part of an Array into Another Array
+
+The `Array.blit` function is available to efficiently copy a portion of an array into another array. Similarly to the `array.(x) <- y`  modification instruction, this function modifies the destination in place and does not return the modified array, but returns `unit`. Let’s suppose you want to copy a part of `even_numbers` into `zeroes`:
+
+```ocaml
+# Array.blit even_numbers 3 zeroes 1 2;;
+- : unit = ()
+# zeroes;;
+- : int array = [|0; 8; 42; 0; 0|]
+```
+
+This copies 2 elements of `even_numbers` starting at index `3` (this array slice is `[| 6; 8 |]`) into `zeroes`, starting at index `1`. It is your responsibility to make sure that the two indices provided are valid in their respective arrays and that the number of elements to copy is within the bounds of each array.
```

Comment:
```suggestion
This copies two elements of `even_numbers`, starting at index `3` (this array slice is `[| 6; 8 |]`) into `zeroes`, starting at index `1`. It is your responsibility to make sure that the two indices provided are valid in their respective arrays and that the number of elements to copy is within the bounds of each array.
```

Reply:
the example given does not follow the logic, when all the modifications of `even_numbers` are fixed this needs to be made to agree